src/System.IdentityModel.Tokens.Jwt/JwtPayload.cs

```diff
@@ -433,12 +432,7 @@ public void AddClaims(IEnumerable<Claim> claims)
                     continue;
                 }
 
-                string jsonClaimType = null;
```

Comment:
I think this is the change you suggested korroz to make, to take out any claim mapping in JwtPayload. The claims that are passed to JwtPayload are already mapped using the outbound claimtype map.